qiime2/sdk/tests/test_result.py

```diff
@@ -646,5 +649,111 @@ def test_invalid_key_added(self):
             collection['not a valid key'] = 0
 
 
+@pytest.fixture
+def signature_test_env(monkeypatch):
+    # fake key info that gpg_find_key would normally parse
+    fake_key_info = {
+        "fingerprint": 'ABCDEF0123456789ABCDEF0123456789ABCDEF01',
+        "algorithm": "Ed25519",
+        "length": 0,
+        "curve": "ed25519",
+        "uids": [{"raw": "Test User <test@example.com>",
+                  "name": "Test User",
+                  "email": "test@example.com"}],
+        "chosen_uid": {"raw": "Test User <test@example.com>",
+                       "name": "Test User",
+                       "email": "test@example.com"}
+    }
+
+    # default behavior will be to succeed
+    key_lookup_behavior = {'mode': 'ok'}
```

Comment:
I think my only question is, is there an implementation reason this is a dict, not just a string? If not, I'd probably just make it a string. Otherwise, this looks good to me. I know the tests feel weak, but we have successfully round-tripped it for real using GPG, and I think we are validating well enough that if GPG works, we work, at least for now.